plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/ConnectionAnnotations.java

```diff
@@ -352,6 +353,35 @@ private ConnectionAnnotations() {
 		throw new UnsupportedOperationException("Helper class must not be instantiated"); //$NON-NLS-1$
 	}
 
+	public static boolean validateNegatedConnection(final Connection connection, final DiagnosticChain diagnostics,
+			final Map<Object, Object> context) {
+		if (connection == null || connection.getSource() == null || connection.getDestination() == null) {
+			return false;
+		}
+
+		if (connection.isNegated()
+				&& (!connection.getSource().getType().equals(IecTypes.ElementaryTypes.BOOL)
+						|| !connection.getDestination().getType().equals(IecTypes.ElementaryTypes.BOOL))
+				&& !hasInOutVar(connection)) {
+			if (diagnostics != null) {
+				diagnostics.add(new BasicDiagnostic(Diagnostic.ERROR, LibraryElementValidator.DIAGNOSTIC_SOURCE,
+						LibraryElementValidator.CONNECTION__VALIDATE_NEGATED_CONNECTION,
+						Messages.ConnectionValidator_NegatedConnectionIsNotValid,
+						FordiacMarkerHelper.getDiagnosticData(connection)));
+			}
+			return false;
+		}
+		return true;
+	}
+
+	private static boolean hasInOutVar(final Connection connection) {
+		if ((connection.getSource() instanceof final VarDeclaration vd && vd.isInOutVar())
+				|| (connection.getDestination() instanceof final VarDeclaration vd2 && vd2.isInOutVar())) {
+			return true;
+		}
+		return false;
```

Comment:
I would suggest to simplify this:
```suggestion
		return (connection.getSource() instanceof final VarDeclaration vd && vd.isInOutVar())
				|| (connection.getDestination() instanceof final VarDeclaration vd2 && vd2.isInOutVar());
```